### PR TITLE
client list displaying in FFA and in CTF mode

### PR DESCRIPTION
**Pull request by weanti**
added client list displaying in FFA and in CTF mode
- client list is displayed only for the host
- client list is scrollable

CTF:
- team distribution: team switching with clicking on a nick
- empty team is not allowed: at least 1 player must be in a list

---
- quit clients are not handled (just like everywhere else)
- multiple clients with the same name cannot be distinguished on the gui

### CODE REVIEW

File: src/network/ctf_client_datas_message.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ void ClientDatasMessageSenderSystem::Init()
 {
     MessageSenderSystem::Init();
         //This one is not used yet you should wire it before use
```
Comment (HalalUr) [inline src/network/ctf_client_datas_message.cpp]: That's not true from now on.

File: src/network/ctf_client_datas_message.cpp [hunk 2]
```diff
@@ -25,6 +26,13 @@ void ClientDatasMessageSenderSystem::Update(double DeltaTime)
...
```
Comment (HalalUr) [inline src/network/ctf_client_datas_message.cpp]: That's not true from now on.

File: src/network/ctf_client_list_system.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+{
+	return mBlueNames;
+}
+
+std::vector<std::string> CtfClientListSystem::redNames()
+{
+    return mRedNames;
+}
+
+void CtfClientListSystem::removeall( core::ProgramState::ClientDatas_t & from , CtfClientListSystem::PlayerClientDataMap const & what )
```
Comment (HalalUr) [inline src/network/ctf_client_list_system.cpp]: Not a big deal but everywhere in the code function names starts with capital.

File: src/network/ctf_client_list_system.cpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#include "ctf_client_datas_message.h"
+#include "ctf_client_datas_changed_event.h"
+#include <algorithm>
+
+namespace network {
+
+using platform::RootModel;
+
+CtfClientListSystem::CtfClientListSystem()
+	: mCTFModel("ctf_client_list", &RootModel::Get())
```
Comment (HalalUr) [inline src/network/ctf_client_list_system.cpp]: I think this is a 'tab'

File: src/network/ctf_client_list_system.cpp (hunk 3 above)
Comment (weanti) [inline src/network/ctf_client_list_system.cpp]: Ok. Consistent style is important. Thanks

File: src/network/ctf_client_list_system.cpp (hunk 4 above)
Comment (weanti) [inline src/network/ctf_client_list_system.cpp]: oh. it is

File: src/network/ctf_client_datas_message.cpp (hunk 1 above)
Comment (weanti) [inline src/network/ctf_client_datas_message.cpp]: Do I need to do anything here (and the other place) besides removing the comment?

Comment (HalalUr) [inline src/network/ctf_client_datas_message.cpp]: Nope, you've added it to main.

**Pull request merged**